Makefile

```diff
@@ -203,7 +203,7 @@ $(CONTROLLER_GEN): $(LOCALBIN)
 .PHONY: envtest
 envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
 $(ENVTEST): $(LOCALBIN)
-	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240208111015-5923139bc5bd
```

Comment:
I don't think this causes a problem,
where does the tests fail?

Reply:
That's exactly what causes the problem:
```
go: downloading sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20240327193027-21368602d84b
go: sigs.k8s.io/controller-runtime/tools/setup-envtest@latest: sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240327193027-21368602d84b requires go >= 1.22.0 (running go 1.21.8; GOTOOLCHAIN=local)
make: *** [Makefile:206: /home/felix/ionos/cluster-api-provider-proxmox/bin/setup-envtest] Error 1
```

Reply:
don't have this problem in my machine, neither in CI

Reply:
I'm also not quite sure if the test is necessary, the install should be a no-op if the package is already installed.

Reply:
If you have golang 1.21 installed and you don't have a `setup-envtest` lying around, the `@latest` it'll download simply doesn't work, because it requires golang 1.22. If you already have  a `setup-envtest`, it keeps on working.
I have no idea what the test environment does, but this issue doesn't need further investigation, it's exactly what it says on the tin. I don't get what we're discussing here.

Reply:
This still works in CI cause it always downloads setup-env.
And I tried to delete it, and it worked.
Maybe something wrong in your machine?

Reply:
How can I successfully communicate that this [commit](https://github.com/kubernetes-sigs/controller-runtime/commit/4c2442e4d743d172a2e0126a841ecd274fffc228) means that you require golang 1.22 which is one larger than golang 1.21 which is the minimum go version we currently support.

Let me give you the side by side:
```
felix@kermit ~/ionos/cluster-api-provider-proxmox $ go version; rm bin/setup-envtest && make test
go version go1.22.1 linux/amd64                                                                
test -s /home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen && /home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen --version | grep -q v0.11.3 || \
GOBIN=/home/felix/ionos/cluster-api-provider-proxmox/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.3
/home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen object paths="./..."  
go fmt ./...                                                                                   
go vet ./...                                                                                   
test -s /home/felix/ionos/cluster-api-provider-proxmox/bin/setup-envtest || GOBIN=/home/felix/ionos/cluster-api-provider-proxmox/bin go install sigs.k8s.io/controller-runtime/tools/setup-env
test@latest                                                                                    
KUBEBUILDER_ASSETS="/home/felix/ionos/cluster-api-provider-proxmox/bin/k8s/1.28.0-linux-amd64" go test ./... -coverprofile cover.out

[...]

felix@kermit ~/ionos/cluster-api-provider-proxmox $ go version; rm bin/setup-envtest && make test
go version go1.21.8 linux/amd64
test -s /home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen && /home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen --version | grep -q v0.11.3 || \
GOBIN=/home/felix/ionos/cluster-api-provider-proxmox/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.3
/home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/felix/ionos/cluster-api-provider-proxmox/bin/controller-gen object paths="./..."
go fmt ./...
go vet ./...
test -s /home/felix/ionos/cluster-api-provider-proxmox/bin/setup-envtest || GOBIN=/home/felix/ionos/cluster-api-provider-proxmox/bin go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
go: sigs.k8s.io/controller-runtime/tools/setup-envtest@latest: sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240403033011-e3285ed1d013 requires go >= 1.22.0 (running go 1.21.8; GOTOOLCHAIN=local)
make: *** [Makefile:206: /home/felix/ionos/cluster-api-provider-proxmox/bin/setup-envtest] Error 1
```

Reply:
from your paste: `go: sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240403033011-e3285ed1d013 requires go >= 1.22.0; switching to go1.22.1`.